src/api.v2/model/Sample.js

```diff
@@ -17,6 +17,44 @@ class Sample extends BasicModel {
     super(sql, tableNames.SAMPLE, sampleFields);
   }
 
+  async getSamples(experimentId) {
+    const { sql } = this;
+
+    const metadataObject = `${replaceNullsWithObject('jsonb_object_agg(key, value)', 'key')} as metadata`;
+
+    const sampleFieldsWithAlias = sampleFields.map((field) => `s.${field}`);
+
+    const metadataQuery = sql.select([...sampleFields, sql.raw(metadataObject)])
+      .from(sql.select([...sampleFieldsWithAlias, 'm.key', 'sm_map.value'])
+        .from({ s: tableNames.SAMPLE })
+        .leftJoin(`${tableNames.METADATA_TRACK} as m`, 's.experiment_id', 'm.experiment_id')
+        .leftJoin(`${tableNames.SAMPLE_IN_METADATA_TRACK_MAP} as sm_map`, 's.id', 'sm_map.sample_id')
+        .where('s.experiment_id', experimentId)
+        .as('mainQuery'))
+      .groupBy(sampleFields)
+      .as('select_metadata');
+
+    const sampleFileFields = ['sample_file_type', 'size', 'upload_status'];
+    const sampleFileFieldsWithAlias = sampleFileFields.map((field) => `sf.${field}`);
+    const fileObjectFormatted = sampleFileFields.map((field) => [`'${field}'`, field]);
+    const sampleFileObject = `jsonb_object_agg(sample_file_type,json_build_object(${fileObjectFormatted})) as files`;
```

Comment:
Just in case: do we need to do the `replaceNullsWithObject` thing here?

Reply:
I dont think so, because these are fields which are required and cannot be null for each sample so we can be sure they are there